src/App.tsx

```diff
@@ -15,48 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+  }
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
+export const App: FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+  const toggleSortType = (type: SortType) => {
+    return sortType === type
+      ? setSortType(SortType.NONE)
+      : setSortType(type);
+  };
```

Comment:
I liked the toggle approach but since handleSortByAlphab and handleSortByLength functions are already created, toggle now seems like a redundant function.

Reply:
agree with @nurdanokcu on this one